graphics/src/Skeleton.cc

```diff
@@ -82,12 +81,13 @@ Skeleton::Skeleton(SkeletonNode *_root)
 //////////////////////////////////////////////////
 Skeleton::~Skeleton()
 {
-  for (auto& kv : this->data->nodes)
+  for (auto &kv : this->data->nodes)
     delete kv.second;
-  for (auto& a : this->data->anims)
+  this->data->nodes.clear();
+
+  for (auto &a : this->data->anims)
     delete a;
-  delete this->data;
-  this->data = NULL;
```

Comment:
keep the last two lines for deleting `this->data` and setting to `nullptr`? It's still a raw ptr in common3

Reply:
Ah, yeah, sorry about that - fixed in https://github.com/ignitionrobotics/ign-common/pull/199/commits/c9a3d7221731a7c96b91f8fa161e536ae44c292e